Sources/Recoverable/Recoverable.swift

```diff
@@ -31,44 +29,74 @@ extension UIView: Recoverable {
         layer.cornerRadius = safeViewState.cornerRadius
         layer.masksToBounds = safeViewState.clipToBounds
         
+        startTransitionIfAvailable()
+        
         if safeViewState.backgroundColor != backgroundColor || forced {
             backgroundColor = safeViewState.backgroundColor
         }
     }
+    
+    fileprivate func startTransitionIfAvailable() {
+        if let transitionType = currentSkeletonConfig?.transition {
+            switch transitionType {
+            case .none:
+                recover()
+            case .fade(duration: let duration):
+                fadeIn(duration: duration)
+            }
+        }
+    }
+    
+    @objc fileprivate func recover() {}
 }
 
-extension UILabel {
+extension UILabel{
+    var labelState: RecoverableTextViewState? {
+        get { return ao_get(pkey: &ViewAssociatedKeys.viewState) as? RecoverableTextViewState }
+        set { ao_setOptional(newValue, pkey: &ViewAssociatedKeys.viewState) }
+    }
+    
     override func saveViewState() {
         super.saveViewState()
-        viewState?.text = text
+        labelState = RecoverableTextViewState(view: self)
     }
     
-    override func recoverViewState(forced: Bool) {
-        super.recoverViewState(forced: forced)
-        text = text == " " || forced ? viewState?.text : text
+    override func recover() {
+        super.recover()
+        textColor = labelState?.textColor
     }
 }
 
-extension UITextView {
+extension UITextView{
+    var textState: RecoverableTextViewState? {
+        get { return ao_get(pkey: &ViewAssociatedKeys.viewState) as? RecoverableTextViewState }
+        set { ao_setOptional(newValue, pkey: &ViewAssociatedKeys.viewState) }
+    }
+    
     override func saveViewState() {
         super.saveViewState()
-        viewState?.text = text
+        textState = RecoverableTextViewState(view: self)
     }
     
-    override func recoverViewState(forced: Bool) {
-        super.recoverViewState(forced: forced)
-        text = text == " " || forced ? viewState?.text : text
```

Comment:
Should we recover `text` too, as we do with `textColor`?

Reply:
Not really necessary. Text is no longer removed when skeleton view is presented like before, so no need to retrieve it.

Reply:
Perfect 👌